### PR TITLE
Assert that a type should never be its own superclass.

### PR DESCRIPTION
**Pull request by practicalswift**
Such a condition will create an endless loop in `lookupVisibleMemberDeclsImpl(…)`.

### CODE REVIEW

**Comment by practicalswift** [comment]
This is the same issue described in #647 but handled with an `assert` instead (as suggested by @slavapestov).

This bug is causing some trouble in the form of unnecessary timeouts when fuzzing SourceKit, so it would be great to fail fast (either via an `assert` as in this PR or via a `return` as in #647) while waiting for the long term fix (improved circularity detection) solving the root cause. 

In short, having #815 or #647 merged would speed up the fuzzing of SourceKit significantly :-)

**Comment by lattner** [comment]
Sounds good!
